src/base-validators/array.test.ts

```diff
@@ -0,0 +1,41 @@
+import validator, { ArrayValidationError } from "./array";
+
+describe("validator", () => {
+  it("can validate and invalidate a array", () => {
+    expect(validator({ type: "array" }, "im an array")).toBeInstanceOf(ArrayValidationError);
+    expect(validator({ type: "array" }, [])).toBe(true);
+    expect(validator({ type: "array" }, {})).toBeInstanceOf(ArrayValidationError);
+    expect(validator({ type: "array" }, true)).toBeInstanceOf(ArrayValidationError);
+    expect(validator({ type: "array" }, false)).toBeInstanceOf(ArrayValidationError);
+    expect(validator({ type: "array" }, null)).toBeInstanceOf(ArrayValidationError);
+    expect(validator({ type: "array" }, 123)).toBeInstanceOf(ArrayValidationError);
+    expect(validator({ type: "array" }, [123, 234, 123234])).toBe(true);
+    expect(validator({ type: "array" }, [123, "string"])).toBe(true);
+  });
+
+  it("minItems", () => {
+    expect(validator({ type: "array", minItems: 2 }, [100,10])).toBe(true);
+    expect(validator({ type: "array", minItems: 2 }, [10])).toBeInstanceOf(ArrayValidationError);
+  });
+
+  it("maxItems", () => {
+    expect(validator({ type: "array", maxItems: 2 }, [10,11])).toBe(true);
+    expect(validator({ type: "array", maxItems: 2 }, [11, 10, 9])).toBeInstanceOf(ArrayValidationError);
+  });
+
+  it("uniqueItems", () => {
```

Comment:
I guess this implies non-deep equality when it comes to uniqueness. Maybe we can look at other validators to see how they handle cases where objects have 'deep equality' but not strict 'referential' equality.